src/include/module/audio/audio_stream.h

```diff
@@ -13,6 +13,11 @@
 #include <stdbool.h>
 #include "../ipc/stream.h"
 
+enum sof_audio_stream_state {
+	STREAM_STATE_INITIAL,			/* Initial state, hw params not configured. */
+	STREAM_STATE_READY,			/* Stream ready, hw params configured */
+};
```

Comment:
does this really have to be tracked per stream / buffer? Wouldn't it be enough to track this in `pipeline->state`?

Reply:
This is currently tracked per stream. I just changed `bool hw_params_configured` to `enum sof_audio_stream_state state` to be able to more precisely define the state. Data is buffered between components, and the end of a stream in the first one doesn't mean that subsequent components don't have data buffered yet.

Reply:
@lgirdwood I think the "stream" here is in the sense of `struct audio_stream` https://github.com/thesofproject/sof/blob/7ed449940c4a8c20cfd91c20265cd33cb3484452/src/include/sof/audio/audio_stream.h#L52, which is described as:
```
 * Audio stream is a circular buffer aware of audio format of the data
 * in the buffer so provides API for reading and writing not only bytes,
 * but also samples and frames.
```

Reply:
Yes, this stream refers to the audio buffer between components. The name may not be the most apt, but it derives, as @lyakh rightly pointed out, from the name of the structure that contains this enum – `audio_stream`.